File: app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.java
package ar.com.wolox.android.example.ui.example;

import ar.com.wolox.android.example.utils.UserSession;
import ar.com.wolox.wolmo.core.presenter.BasePresenter;

import javax.inject.Inject;

public class ExamplePresenter extends BasePresenter<IExampleView> {

    // Constants
    public static final String TAG = "ExamplePresenter";
    public static final int NUMBER_MAX = 1000;
    public static final int NUMBER_MIN = 200;

    // Variables
    private UserSession mUserSession;

    // Constructor
    @Inject
    public ExamplePresenter(UserSession userSession) {
        mUserSession = userSession;
    }

    public void storeUsername(String text) {
        mUserSession.setUsername(text);
        getView().onUsernameSaved();
    }
}

File: app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.kt
package ar.com.wolox.android.example.ui.example

import ar.com.wolox.android.example.utils.UserSession
import ar.com.wolox.wolmo.core.presenter.BasePresenter
import javax.inject.Inject

class ExamplePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<ExampleView>() {

    fun onLoginButtonClicked(user: String, color: String) {
        userSession.username = user
        view?.goToViewPager(color)
    }

    fun onUsernameInputChanged(text: String) = view?.toggleLoginButtonEnable(text.isNotBlank())

    fun onWoloxLinkClicked() = view?.openBrowser(WOLOX_URL)

    fun onWoloxPhoneClicked() = view?.openPhone(WOLOX_PHONE)

    companion object {
        private const val WOLOX_URL = "www.wolox.com.ar"
        private const val WOLOX_PHONE = "08001234567"
    }
}
